cloudify_gcp/utils.py

```diff
@@ -124,6 +130,18 @@ def get_final_resource_name(name):
 def create_resource(func):
     def _decorator(resource, *args, **kwargs):
         if should_use_external_resource():
+            if not ctx.instance.runtime_properties.get(constants.RESOURCE_ID):
+                resource_id = ctx.node.properties.get(constants.RESOURCE_ID)
+                if not resource_id:
+                    name = ctx.node.properties.get('name')
+                    if name:
+                        resource_id = name
+                        ctx.instance.runtime_properties[
+                            constants.RESOURCE_ID] = resource_id
+
```

Comment:
Perhaps we should add a check that if someone provides both `resource_id` and `name` and the values are not the same.

Reply:
ok, in this case should I raise error?

Reply:
yes

Reply:
@GhalebKhaled I think we also need to check if both `name` and `resource_id` is not provided because there are some resources that do not have a `name` prop, on that case we need to check if both are empty then we should raise error

Reply:
already done in the line below :)